patternfly/patternfly-elements
# fix: Typography: mixins & extends

## Component name

- pfe-sass
- pfe-styles

  
### What has changed and why

- Move towards mixins instead of extends for  CSS performance reasons. Left the placeholders in tact, but they lean on the mixins.
- Update sassdocs for typography helpers
- Fix variables so that `--pfe-c-content` only has one dash between
- Fix names for text modifier classes (previously there was a mismatch between var names & classes, one said "title", now they match)
![image](https://user-images.githubusercontent.com/456863/106508364-4fb5fc80-649a-11eb-8666-80c6cf880bed.png)
- Update typography mixins to allow for argument that allows opt-in printing of !important on utility class properties (but not extends or standard use of the mixin)
- Combined title & text modifiers into one map


### Testing instructions

1.  pfe styles page should look the same as the master branch
2. Variable names should match [source of truth typography spreadsheet](https://docs.google.com/spreadsheets/d/1OnCRfYH9IduVLSGAVHki9xQBZSegHOcIZeXFONmAKNw/edit#gid=482567725)

- [Sassdoc page](https://deploy-preview-1303--happy-galileo-ea79c4.netlify.app/elements/pfe-sass/demo/#typography-mixin-pfe-title)
- [Styles page](https://deploy-preview-1303--happy-galileo-ea79c4.netlify.app/elements/pfe-styles/demo)


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable.



### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
